src/14_legendre.jl

```diff
@@ -0,0 +1,418 @@
+    # Legendre polynomial P_n(x) can be evaluated outside of [-1,1].
+    # Thus, we use that routine before checking x interval for associated Legendre polynomial P_n^m(x).
+    if 0 == m
+        return legendrep(n, x)
+    end
+
+    # general argument checks
+    if n < 0
+        throw(DomainError(n, "must be non-negative"))
+    end
+    if m < 0 || m > n
+        throw(DomainError(m, "must be in the range m = 0,..,n"))
+    end
+    if x < -1 || x > 1
+        throw(DomainError(x, "must be in the range [-1,1]"))
+    end
+
+    # step 1: compute P_m^m(x)
+    p = 1                                           # = P_0^0(x)
+    for k = 1:m
+        p_prev  = p
+        p       = -(2k-1) * sqrt(1-x^2) * p_prev    # = P_k^k(x)
+    end                                             # on output: p = P_m^m(x)
+    if n == m
+        return p
+    end
+
+    # step 2: compute P_{m+1}^m(x)
+    p_prev  = p                                     # = P_m^m(x)
+    p       = (2m+1) * x * p_prev                   # = P_{m+1}^m(x)
+
+    # step 3: compute P_n^m(x)
+    for k = m+2:n
+        p_prev, p_prev_prev = p, p_prev
+        p = (2k-1)/(k-m)*x*p_prev - (k+m-1)/(k-m)*p_prev_prev       # = P_k^m(x)
+    end                                             # on output: p = P_n^m(x)
+
+    p
+end
+
+@doc raw"""
+    legendreq(n[, m], x)
+
+Evaluate the Legendre function of second kind ``Q_n(x)`` of integer degree ``n`` at position
+``x``.
+If the order ``m`` is supplied then the associated Legendre function of second kind
+``Q_n^{(m)}(x)`` is computed.
+Their definitions are given by
+
+```math
+\begin{align}
+    Q_n^{(m)}(x)
+    &=
+    (-1)^m (1-x^2)^\frac{m}{2} \frac{\mathrm{d}^m}{\mathrm{d}x^m} \left( Q_n(x) \right)
+    \,, \quad m \in \mathbb{N}_0 \,, \; x \in (-1,1)
+    \\
+    Q_n(x)
+    &=
+    \begin{cases}
+        \frac{1}{2} \log \frac{1+x}{1-x}
+        & \text{for} \quad n = 0
+        \\
+        P_1(x) Q_0(x) - 1
+        & \text{for} \quad n = 1
+        \\
+        \frac{2n-1}{n} x Q_{n-1}(x) - \frac{n-1}{n} Q_{n-2}(x)
+        & \text{for} \quad n \geq 2
+        \,.
+    \end{cases}
+\end{align}
+```
+where ``P_1`` is the Legendre polynomial of first kind and degree one.
+The functions diverge at the interval boundaries
+``\lim_{x\rightarrow \pm 1} |Q_n(x)| = \infty``.
+
+External links: [DLMF - Legendre function](https://dlmf.nist.gov/14.7.E2),
+[DLMF - associated Legendre function](https://dlmf.nist.gov/14.7.E9),
+[Wikipedia - Legendre function](https://en.wikipedia.org/wiki/Legendre_functions#Legendre_functions_of_the_second_kind_(Qn)),
+[Wikipedia - associated Legendre function](https://en.wikipedia.org/wiki/Legendre_function#Associated_Legendre_functions_of_the_second_kind).
+"""
+function legendreq(n::Integer, x::Real)
+    if n < 0
+        throw(DomainError(n, "must be non-negative"))
+    end
+    if x < -1 || x > 1
+        throw(DomainError(x, "must be in the range [-1,1]"))
+    end
+
+    if      x == 1
```

Comment:
```suggestion
    if  x == 1
```

Reply:
I favoured this style to align the conditions of the <code>if</code> and <code>elseif</code> lines. Will change it in a bit..

Reply:
done. see commit [4ab8c7f](https://github.com/JuliaMath/SpecialFunctions.jl/pull/175/commits/4ab8c7f3c88777c020c594eb5335a38d44ab075d)